src/adapters/github.ts

```diff
@@ -50,12 +50,55 @@ class GithubAdapter extends GitAdapter {
       }
     }
   }
-
+  // private async getOrgRepos(org: string, onRetry: RetryMethod): Promise<IRepo[]> {
+  //     const repos = await paginate(this.octokit, this.octokit.repos.listForOrg, undefined, onRetry)({
+  //       org,
+  //     });
+  //     const repoNames = repos.map((r: any) => r.full_name).sort();
+  //     return _.uniq(repoNames).map((r: string) => this.parseRepo(r));
+  // }
+
+  // private async getCodeSearchRepos(search_query: string, onRetry: RetryMethod): Promise<IRepo[]> {
+  //   const searchResults = await paginateSearch(this.octokit, this.octokit.search.code, onRetry)({
+  //     q: search_query,
+  //   });
+  //   const repoNames = searchResults.map((r: any) => r.repository.full_name).sort();
+  //   return _.uniq(repoNames).map((r: string) => this.parseRepo(r));
+  // }
+
+  // public async getCandidateRepos(onRetry: RetryMethod): Promise<IRepo[]> {
+  //   const { org, search_query } = this.migrationContext.migration.spec.adapter;
+  //   // list all of an orgs repos
+  //   if (org) {
+  //     if (search_query) {
+  //       throw new Error('Cannot use both "org" and "search_query" in github adapter. Pick one.');
+  //     }
+  //      return this.getOrgRepos(org, onRetry);
+  //   }
+  //   // otherwise, github code search query.  results are less reliable
+  //   return this.getCodeSearchRepos(search_query, onRetry);
+  // }
   public async getCandidateRepos(onRetry: RetryMethod): Promise<IRepo[]> {
-    const searchResults = await paginateSearch(this.octokit, this.octokit.search.code, onRetry)({
-      q: this.migrationContext.migration.spec.adapter.search_query,
-    });
-    const repoNames = searchResults.map((r: any) => r.repository.full_name).sort();
+    const { org, search_query } = this.migrationContext.migration.spec.adapter;
+    let repoNames = [];
+
+    // list all of an orgs repos
+    if (org) {
+      if (search_query) {
+        throw new Error('Cannot use both "org" and "search_query" in github adapter. Pick one.');
+      }
+      const repos = await paginate(this.octokit, this.octokit.repos.listForOrg, undefined, onRetry)({
```

Comment:
Nit: I think this is outside the scope of this PR but just wanted pose the question 💭 

Perhaps in the future we could refactor [the signature of `paginate`](https://github.com/NerdWalletOSS/shepherd/blob/master/src/util/octokit.ts#L12) so that at the call site we could omit `extractorItems`. And it'd have more symmetry with `paginateSearch`.

Reply:
+1 - keeping optional parameters at the end of the parameter list seems sensible!

Reply:
+1 - could also go for an interface that takes a single argument object

That's for a different PR for a different time 👍